packages/fxa-settings/src/models/reliers/factory.ts

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { useContext } from 'react';
+import { DefaultRelierFlags, RelierFactory } from '../../lib/reliers';
+import { AppContext } from '../AppContext';
+
+export function CreateRelierDelegates() {
+  const { oauthClient, authClient } = useContext(AppContext);
+
+  if (!oauthClient || !authClient) {
+    throw new Error('Are you forgetting an AppContext.Provider?');
+  }
+
+  const delegates = {
+    getClientInfo: (id: string) => oauthClient.getClientInfo(id),
+    getProductInfo: (id: string) => authClient.getProductInfo(id),
+    getProductIdFromRoute: () => {
+      const re = new RegExp('/subscriptions/products/(.*)');
+      return re.exec(window.location.pathname)?.[1] || '';
```

Comment:
Is this the global `window`?  Do we need some sort of wrapper here?

Reply:
Oh good catch. I missed this one. Let me swap it out.